src/org/thoughtcrime/securesms/util/Prefs.java

```diff
@@ -191,36 +191,54 @@ public static boolean isLocationStreamingEnabled(Context context) {
 
   // ringtone
 
-  public static @NonNull Uri getNotificationRingtone(Context context) {
-    String result = getStringPreference(context, RINGTONE_PREF, Settings.System.DEFAULT_NOTIFICATION_URI.toString());
-
-    if (result != null && result.startsWith("file:")) {
-      result = Settings.System.DEFAULT_NOTIFICATION_URI.toString();
+  public static @Nullable Uri getNotificationRingtone(Context context) {
+    String result = getStringPreference(context, RINGTONE_PREF, null);
+    if (result == null || result.startsWith("file:")) {
+      return Settings.System.DEFAULT_NOTIFICATION_URI;
+    }
+    if (result.isEmpty()) {
+      return null;
     }
-
     return Uri.parse(result);
   }
 
-  public static void removeNotificationRingtone(Context context) {
-    removePreference(context, RINGTONE_PREF);
-  }
-
   public static void setNotificationRingtone(Context context, Uri ringtone) {
-    setStringPreference(context, RINGTONE_PREF, ringtone.toString());
+    if (Settings.System.DEFAULT_NOTIFICATION_URI.equals(ringtone)) {
+      removePreference(context, RINGTONE_PREF);
+    } else {
+      if (ringtone == null) {
+        ringtone = Uri.EMPTY;
+      }
+      setStringPreference(context, RINGTONE_PREF, ringtone.toString());
+    }
   }
 
   public static void setChatRingtone(Context context, int chatId, Uri ringtone) {
-    if(ringtone!=null) {
-      setStringPreference(context, CHAT_RINGTONE+chatId, ringtone.toString());
-    }
-    else {
+    Uri defaultValue = getNotificationRingtone(context);
+    if ((defaultValue == null && ringtone == null)
+        || (defaultValue != null && defaultValue.equals(ringtone))) {
       removePreference(context, CHAT_RINGTONE+chatId);
+    } else {
+      if (ringtone == null) {
+        ringtone = Uri.EMPTY;
+      }
+      setStringPreference(context, CHAT_RINGTONE+chatId, ringtone.toString());
     }
   }
 
+  public static boolean hasDefaultRingtone(Context context, int chatId) {
```

Comment:
> however, tbh, i do not really get why we need two functions at all.

because java sucks, in python I could just `getChatRingtone() == getNotificationRingtone()` but IIRC in java you can't compare two strings with `==` right? I would need to check first if one is null then compare with `==` or with `string.equals()`, so I did it this way which is optimal

Reply:
That's what we have Util.equals for, you can do `Util.equals(getChatRingtone(), getNotificationRingtone())`

Reply:
still not sure if it is good to call that two functions that will be called in `addNotification()` in each incoming message, and it is a bit unnecessary, when all it is needed to know is if the stored preference doesn't exist (`==null`)